### PR TITLE
EVA-2586 - Script to gather FTP usage stats

### PR DESCRIPTION
**Pull request by apriltuesday**
I've also included the script for web services usage with no changes except minimal stylistic ones.

### CODE REVIEW

File: eva-usage-stats/ftp_usage.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            "select max(event_ts_txt) as recent_ts from eva_web_srvc_stats.ftp_traffic;"
+        )
+        if results and results[0][0]:
+            return results[0][0]
+    return None
+
+
+@retry(tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def query(kibana_host, basic_auth, private_config_xml_file):
+    first_query_url = os.path.join(kibana_host, 'ftplogs*/_search?scroll=24h')
```
Comment (tcezard) [inline eva-usage-stats/ftp_usage.py:82]: I think this query will be fine but we might get a faster query if we limit the search space to a single (or a few) index in line with the `most_recent_timestamp`. We can revisit if we need to.

**Review by tcezard — APPROVED** [review]

File: eva-usage-stats/ftp_usage.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    response.raise_for_status()
+    data = response.json()
+    return data['hits']['hits']
+
+
+def main():
+    parser = ArgumentParser(description='Retrieves data from Kibana and dumps into a local postgres instance')
+    parser.add_argument('--kibana-host', help='Kibana host to query, e.g. http://example.ebi.ac.uk:9200', required=True)
+    parser.add_argument('--kibana-user', help='Kibana API username', required=True)
+    parser.add_argument('--kibana-pass', help='Kibana API password', required=True)
```
Comment (sundarvenkata-EBI) [inline eva-usage-stats/ftp_usage.py:116]: Is this a common password or user-specific? If it is the latter, I am not sure if we should expose it on the command line...

Comment (tcezard) [reply eva-usage-stats/ftp_usage.py:116]: It is the same (weak) password for everyone

Comment (sundarvenkata-EBI) [reply eva-usage-stats/ftp_usage.py:116]: :+1:

File: eva-usage-stats/ftp_usage.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                '%s, %s, %s, %s, %s, %s, %s)'
+            )
+            psycopg2.extras.execute_batch(cursor, query_insert, rows)
+
+
+def get_most_recent_timestamp(private_config_xml_file):
+    with get_metadata_connection_handle('development', private_config_xml_file) as metadata_connection_handle:
+        results = get_all_results_for_query(
+            metadata_connection_handle,
+            "select max(event_ts) as recent_ts from eva_web_srvc_stats.ftp_traffic;"
```
Comment (sundarvenkata-EBI) [inline eva-usage-stats/ftp_usage.py:72]: Suggest refactoring the table name to a variable

**Review by sundarvenkata-EBI — APPROVED** [review]
Looks good!

**Pull request merged**